src/ray/raylet/worker_pool.cc

```diff
@@ -1177,13 +1186,14 @@ void WorkerPool::PopWorker(const TaskSpecification &task_spec,
       } else {
         state.starting_workers_to_tasks[startup_token] = std::move(task_info);
       }
+    } else if (status == PopWorkerStatus::TooManyStartingWorkerProcesses) {
+      DeleteRuntimeEnvIfPossible(task_spec.SerializedRuntimeEnv());
```

Comment:
Actually, the ideal way is that we don't `DeleteRuntimeEnvIfPossible` here and reuse it when pop worker next time, right?

Reply:
Yea, I agree. I'll add a TODO here and address it in the follow-up PR since it's not a regression. Does that sound good to you?

Reply:
Good for me.